cmake/find/FindFilesystem.cmake

```diff
@@ -236,13 +240,40 @@ if (NOT _found AND "Boost" IN_LIST want_components)
     find_package(Boost COMPONENTS filesystem)
     if (TARGET Boost::filesystem)
         set(_found TRUE)
-        set(CXX_FILESYSTEM_HEADER boost/filesystem.hpp CACHE STRING "The header that should be included to obtain the filesystem APIs")
-        set(CXX_FILESYSTEM_NAMESPACE boost::filesystem CACHE STRING "The C++ namespace that contains the filesystem APIs")
+        # Bug fix by Bryan Tan
+        set(CXX_FILESYSTEM_HEADER boost/filesystem.hpp CACHE STRING "The header that should be included to obtain the filesystem APIs" FORCE)
+        set(CXX_FILESYSTEM_NAMESPACE boost::filesystem CACHE STRING "The C++ namespace that contains the filesystem APIs" FORCE)
+        # End bug fix
         add_library(std_filesystem INTERFACE)
         target_compile_features(std_filesystem INTERFACE cxx_std_17)
         target_link_libraries(std_filesystem INTERFACE Boost::boost Boost::filesystem)
+        target_include_directories(std_filesystem INTERFACE ${Boost_INCLUDE_DIRS})
         add_library(std::filesystem ALIAS std_filesystem)
     endif ()
+elseif (NOT _found AND "ghc" IN_LIST want_components)
+    message("-- Using ghc::filesystem (git download)")
+    set (GHC_FILESYSTEM ghcFilesystem_git)
+    ExternalProject_Add(${GHC_FILESYSTEM}
+      GIT_REPOSITORY "https://github.com/gulrak/filesystem.git"
+      GIT_TAG "v1.3.2"
+      LOG_DOWNLOAD 1
+      CONFIGURE_COMMAND ""
+      BUILD_COMMAND ""
+      UPDATE_COMMAND ""
+      INSTALL_COMMAND ""
+      )
+    set(_found TRUE)
+    set(CXX_FILESYSTEM_HEADER ghc/filesystem.hpp CACHE STRING "The header that should be included to obtain the filesystem APIs" FORCE)
+    set(CXX_FILESYSTEM_NAMESPACE ghc::filesystem CACHE STRING "The C++ namespace that contains the filesystem APIs" FORCE)
+
+    add_library(std_filesystem INTERFACE)
+    add_dependencies(std_filesystem ghcFilesystem_git)
+    target_compile_features(std_filesystem INTERFACE cxx_std_17)
+    ExternalProject_Get_Property(${GHC_FILESYSTEM} SOURCE_DIR)
+    file(MAKE_DIRECTORY "${SOURCE_DIR}/include")
+    target_include_directories(std_filesystem INTERFACE "${SOURCE_DIR}/include")
+
+    add_library(std::filesystem ALIAS std_filesystem)
```

Comment:
I think this could be simplified, since ExternalProject_Add creates a target, containing all dependencies.

Reply:
`ExternalProject_Add` only builds and installs the subproject; it does _not_ import the targets of the subproject. The preferred option is [`FetchContent`](https://cmake.org/cmake/help/git-stage/module/FetchContent.html) but it is only available on CMake 3.11+ (we unfortunately require 3.10).